cypress/integration/core/header.spec.ts

```diff
@@ -1,6 +1,59 @@
 describe("Header", () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:56106/iframe.html?id=header--header');
+  });
+
+  const defaultMenuItems = [
+    {
+      label: "Home",
+      url: "#home",
+      active: true,
+    },
+    {
+      label: "Vergunningscheck",
+      url: "#vergunningscheck",
+    },
+    {
+      label: "Aanvragen",
+      url: "#aanvragen",
+    },
+    {
+      label: "Regels op de kaart",
+      url: "#regelsopdekaart",
+    },
+    {
+      label: "Maatregelen op maat",
+      url: "#maatregelenopmaat",
+    },
+    {
+      label: "Hulpcentrum",
+      url: "#hulpcentrum",
+    },
+  ];
+
+  /** Configure the component and set an eventListener as @headerListener the `dso-header` is set as @dsoHeader and the `dso-header` shadow dom as @dsoHeaderShadow */
+  function prepareComponent() {
+    cy.get('dso-header')
+      .as('dsoHeader')
+      .then($header => {
+        $header.on('headerClick', ($event: any) => {
+          $event.detail.originalEvent.preventDefault();
+        });
+        $header.on('headerClick', cy.stub().as('headerListener'));
+      })
+      .shadow()
+      .as('dsoHeaderShadow');
+  }
+
+  function setMenuItems($header: JQuery<HTMLElement>, items?: any[]) {
+    const element = $header.get(0);
+
+    if (element) {
+      (element as any).mainMenu = items;
```

Comment:
dit is een beetje jammer, ik had liever `if(element instanceof HTMLDsoHeaderElement)` willen doen maar dat gaat helaas niet.